### PR TITLE
Feature move 96 hente printinfo krr

### PR DESCRIPTION
**Pull request by mhoel**

### CODE REVIEW

**Comment by nuth** [comment]
build

File: kontakt-reservasjonsregister-client/src/main/java/no/difi/meldingsutveksling/ptp/PersonKontaktInfoMapper.java [hunk 1]
```diff
@@ -1,35 +1,163 @@
...
+        public boolean isAktiv() {
+            return aktiv;
+        }
+
+        /**
+         *
+         * @return true if all necessary details are in place for sending digital post
+         */
+        public boolean isEmpty() {
+            return (emailAdress == null && phoneNumber == null);
```
Comment (nuth) [inline kontakt-reservasjonsregister-client/src/main/java/no/difi/meldingsutveksling/ptp/PersonKontaktInfoMapper.java]: MAJOR Remove those useless parentheses. [rule](http://sonar-app01.difi.local:9000/coding_rules#rule_key=squid%3AUselessParenthesesCheck)

File: kontakt-reservasjonsregister-client/src/main/java/no/difi/meldingsutveksling/ptp/PersonKontaktInfoMapper.java [hunk 2]
```diff
@@ -1,35 +1,163 @@
...
+
+    private static boolean hasKontaktInformasjon(Person person) {
+        return person.getKontaktinformasjon() != null;
+    }
+
+    private static boolean hasPostkasseleverandoerAdresse(Person person) {
+        return person.getSikkerDigitalPostAdresse() != null;
+    }
+
+    private static boolean hasCertificate(Person person) {
```
Comment (nuth) [inline kontakt-reservasjonsregister-client/src/main/java/no/difi/meldingsutveksling/ptp/PersonKontaktInfoMapper.java]: MAJOR Remove this unused private "hasCertificate" method. [rule](http://sonar-app01.difi.local:9000/coding_rules#rule_key=squid%3AUnusedPrivateMethod)

File: kontakt-reservasjonsregister-client/src/main/java/no/difi/meldingsutveksling/ptp/PostAddress.java [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    private final String name;
+    private String street1;
+    private String street2;
+    private String street3;
+    private String street4;
+    private final String postalCode;
+    private final String postalArea;
+    private final String country;
+
+    public PostAddress(String name, String street1, String street2, String street3, String street4, String postalCode, String postalArea, String country) {
```
Comment (nuth) [inline kontakt-reservasjonsregister-client/src/main/java/no/difi/meldingsutveksling/ptp/PostAddress.java]: MAJOR Constructor has 8 parameters, which is greater than 7 authorized. [rule](http://sonar-app01.difi.local:9000/coding_rules#rule_key=squid%3AS00107)

**Comment by nuth** [comment]
SonarQube analysis reported 9 issues
* CRITICAL 3 critical
* MAJOR 6 major

Watch the comments in this conversation to review them.

File: kontakt-reservasjonsregister-client/src/main/java/no/difi/meldingsutveksling/ptp/Util.java [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+import org.bouncycastle.openssl.jcajce.JcaPEMWriter;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.StringWriter;
+import java.security.cert.CertificateException;
+import java.security.cert.CertificateFactory;
+import java.security.cert.X509Certificate;
+
+class Util {
```
Comment (martinwam) [inline kontakt-reservasjonsregister-client/src/main/java/no/difi/meldingsutveksling/ptp/Util.java]: Litt snaut å kalle en klasse *bare* "Util"?

**Pull request closed without merging**